tests/MutantTest.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+/**
+ *
+ * @category   Humbug
+ * @package    Humbug
+ * @copyright  Copyright (c) 2015 Pádraic Brady (http://blog.astrumfutura.com)
+ * @license    https://github.com/padraic/humbug/blob/master/LICENSE New BSD License
+ * @author     Thibaud Fabre
+ */
+namespace Humbug\Test;
+
+use Humbug\Exception\NoCoveringTestsException;
+use Humbug\Mutant;
+use Humbug\Mutation;
+use Humbug\TestSuite\Mutant\FileGenerator;
+use Humbug\Utility\CoverageData;
+use Prophecy\Argument;
+
+class MutantTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @return FileGenerator
+     */
+    public function getFileGenerator()
+    {
+        $generator = $this->prophesize('Humbug\TestSuite\Mutant\FileGenerator');
+        $generator->generateFile(Argument::type('Humbug\Mutation'))
+            ->willReturn(__DIR__ . '/_files/mutants/mutant.pĥp');
+
+        return $generator->reveal();
+    }
+
+    /**
+     * @param bool $throwException
+     *
+     * @return CoverageData
+     */
+    public function getCoverageData($throwException = false, array $tests = [], array $testMethods = [])
+    {
+        $coverageData = $this->prophesize('Humbug\Utility\CoverageData');
+        $getTestMethod = $coverageData->getTestClasses(Argument::any(), Argument::any());
+
+        if ($throwException) {
```

Comment:
It is hard to maintain tests with logic.
I would consider splitting into two separate methods: `getCoverageDataWithTests` and `getCoverageDataRisingExceptionDuringGetTestClasses`